Simulation.jl

```diff
@@ -0,0 +1,50 @@
+function simulate_deposition(fcoeffs, Ea_values, T_range, num_steps, dt)
+    # Initialize existing_layers as a 2D array
+    new_layer = [1.0 0.0 0.0 0.0 0.0 0.0]
+    compositions_all = [] # Initialize compositions for all temperatures
+    A = 0.1 # Arrhenius prefactor 
+    gprob = 1.0 # Probability of a new layer being deposited
+    first_run = true
+    for fcoeff in fcoeffs
+        println("1")
+        compositions_T = [] # Initialize compositions for a single temperature
+        for T in T_range
+            existing_layers = [1.0 0.0 0.0 0.0 0.0 0.0] # Initial layer(Amorphous)
+            k_m = arrhenius_rate_matrix(Ea_values, T) # Calculate rate constants for all phases
+            #display(k_m)
+            for _ in 1:num_steps
+                e = copy(existing_layers)
+                for i in 1:size(existing_layers, 1)
+                    e[i, :] = fcoeff[i] * existing_layers[i, :] # for more inner layers, multiply by smaller fcoeff
```

Comment:
we should write this in a way that the `e` array doesn't change size during evaluation. Precompute what its size is an initialize accordingly, then just populate as you go.

Reply:
also, this step should be able to be done as an elementwise multiply without having to write out the loop, much easier to read.